duckduckgo/iOS
# Add Sync feature flags

Task/Issue URL: https://app.asana.com/0/0/1206046777189407/f

**Description**:
This change add support for Sync feature in Privacy Configuration, together with
4 subfeatures defining availability of various parts of Sync experience.
DDGSyncing gets a read-only feature flag variable as well as a publisher.
PrivacyConfigurationManager is now a Sync dependency, and DDGSync takes
care internally of listening to Privacy Config changes and updating feature flags
as needed.
Feature flag responsible for actual data syncing is handled internally in DDGSync
by cancelling all pending sync operations and disabling adding new operations.
Other feature flags should be handled by client apps.

**Steps to test this PR**:
1. Fetch existing privacy config files for iOS and macOS.
2. Serve them from a different location.
3. Run each app, update Privacy Config URL (for macOS this can be done in runtime via Debug Menu).
4. Manipulate feature flags and verify that they work according to the specification.



**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
